packages/@aws-cdk/pipelines/lib/actions/deploy-cdk-stack-action.ts

```diff
@@ -156,12 +162,25 @@ export class DeployCdkStackAction implements codepipeline.IAction {
     // It should be easier to get this, but for now it is what it is.
     const appAsmRoot = assemblyBuilderOf(appOf(scope)).outdir;
     const fullTemplatePath = path.join(artifact.assembly.directory, artifact.templateFile);
-    const templatePath = path.relative(appAsmRoot, fullTemplatePath);
+
+    let fullConfigPath;
+    if (Object.keys(artifact.tags).length > 0) {
+      fullConfigPath = `${fullTemplatePath}.config.json`;
+
+      // Write the template configuration file (for parameters into CreateChangeSet call that
+      // cannot be configured any other way). They must come from a file, and there's unfortunately
```

Comment:
What's wrong with this way?

Reply:
Feels like this would be an excellent case for a construct adding an artifact to the synthesized output of a cloud assembly, as an implementation detail of the construct.

What's happening now is that we're writing a file which is right next to another file that we *assume* is going to be in a cloud assembly, of which we kinda guess the name is probably going to be unused, which is invisible any of the rest of the framework.